### ISSUE TITLE
Incompatibility with numba 0.50.1

### ISSUE THREAD

**Issue by syoffe** [fbpic/fbpic#458]
It seems that `numba=0.50.1` breaks compatibility with `fbpic=0.16.0`. For now, downgrading to `numba=0.49.1` is a temporary fix for the issue. Here is the traceback for the `numba.cuda.cudadrv.error.NvvmError: Failed to compile` error encountered during the JIT compilation:
```
Just-In-Time compilation (up to one minute) ...Traceback (most recent call last):
  File "fbpic_script.py", line 150, in <module>
    sim.step( N_step )
  File ".../miniconda/lib/python3.7/site-packages/fbpic/main.py", line 429, in step
    self.deposit('rho_prev', exchange=(use_true_rho is True))
  File ".../miniconda/lib/python3.7/site-packages/fbpic/main.py", line 611, in deposit
    species.deposit( fld, 'rho' )
  File ".../miniconda/lib/python3.7/site-packages/fbpic/particles/particles.py", line 912, in deposit
    self.cell_idx, self.prefix_sum)
  File ".../miniconda/lib/python3.7/site-packages/fbpic/utils/cuda.py", line 319, in call_kernel
    .specialize(*args)
  File ".../miniconda/lib/python3.7/site-packages/numba/cuda/compiler.py", line 844, in specialize
    kernel = self.compile(argtypes)
  File ".../miniconda/lib/python3.7/site-packages/numba/cuda/compiler.py", line 863, in compile
    kernel.bind()
  File ".../miniconda/lib/python3.7/site-packages/numba/cuda/compiler.py", line 604, in bind
    self._func.get()
  File ".../miniconda/lib/python3.7/site-packages/numba/cuda/compiler.py", line 480, in get
    ptx = self.ptx.get()
  File ".../miniconda/lib/python3.7/site-packages/numba/cuda/compiler.py", line 451, in get
    **self._extra_options)
  File ".../miniconda/lib/python3.7/site-packages/numba/cuda/cudadrv/nvvm.py", line 515, in llvm_to_ptx
    ptx = cu.compile(**opts)
  File ".../miniconda/lib/python3.7/site-packages/numba/cuda/cudadrv/nvvm.py", line 232, in compile
    self._try_error(err, 'Failed to compile\n')
  File ".../miniconda/lib/python3.7/site-packages/numba/cuda/cudadrv/nvvm.py", line 250, in _try_error
    self.driver.check_error(err, "%s\n%s" % (msg, self.get_log()))
  File ".../miniconda/lib/python3.7/site-packages/numba/cuda/cudadrv/nvvm.py", line 140, in check_error
    raise exc
numba.cuda.cudadrv.error.NvvmError: Failed to compile

<unnamed> (359, 17): parse expected binary operation in atomicrmw
NVVM_ERROR_COMPILATION
```

**Comment by RemiLehe** [comment]
Thanks for reporting this issue!
This issue (somewhat more isolated) was reported on the Numba repository:
https://github.com/numba/numba/issues/5968
For now, it seems that another temporary workaround is to install `numba` and `llvmlite` from `pip`.

**Comment by RemiLehe** [comment]
The issue has been fixed on the `numba/llvmlite` side. Downloading `numba` version 0.50.1 from `conda` works again with FBPIC. 

However, note that in the latest release of FBPIC, I added a preemptive check that prevents users to run with `numba` version 0.50.1. (This was released before the above-mentioned fix in `numba/llvmlite`.) This has been removed in the `dev` branch of FBPIC, but will only be available with the next release.
